File: growing-shop-auth-before/src/main/java/com/example/growingshopauth/auth/repository/RoleRepository.java
package com.example.growingshopauth.auth.repository;

import com.example.growingshopauth.auth.domain.Role;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface RoleRepository extends JpaRepository<Role, Long> {
    Optional<Role> getByName(String name);
}

File: growing-shop-auth/src/main/kotlin/com/example/growingshopauth/auth/repository/RoleRepository.kt
package com.example.growingshopauth.auth.repository

import com.example.growingshopauth.auth.domain.Role
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface RoleRepository: JpaRepository<Role, Long> {

    fun getByName(name: String): Optional<Role>
}
